packages/react-storage/src/components/StorageBrowser/hooks/useSortHandler.ts

```diff
@@ -0,0 +1,96 @@
+import { useCallback, useState } from 'react';
+import { Column, ILocation, SortDirection } from '../Views/Controls/Table';
+
+export interface CompareFnType {
+  (a: any, b: any): number;
+}
+
+function CompareFn(a: string, b: string): number;
+function CompareFn(a: number, b: number): number;
+function CompareFn(a: string | number, b: string | number): number {
+  const isDateString = (s: string) => !isNaN(Date.parse(s));
+
+  if (typeof a === 'string' && typeof b === 'string') {
+    if (isDateString(a) && isDateString(b)) {
+      // Compare strings as dates
+      return new Date(a).getTime() - new Date(b).getTime();
+    }
+
+    // Otherwise, compare them as regular strings
+    return a.localeCompare(b);
+  } else if (typeof a === 'number' && typeof b === 'number') {
+    // Compare numbers
+    return a - b;
+  }
+  throw new Error('Invalid arguments');
+}
+
+interface ISortState<T> {
```

Comment:
Why `ISortState`, none of our existing types have the `I` prefix?

Reply:
ah I can remove the `I`. I previously exported this interface and thought it was kind of confusing since I had `sortState` being used